pandas/io/packers.py

```diff
@@ -203,18 +202,21 @@ def read(fh):
             with open(path_or_buf, 'rb') as fh:
                 return read(fh)
 
-    # treat as a binary-like
     if isinstance(path_or_buf, compat.binary_type):
+        # treat as a binary-like
         fh = None
         try:
-            fh = compat.BytesIO(path_or_buf)
-            return read(fh)
+            # We can't distinguish between a path and a buffer of bytes in
+            # Python 2 so instead assume the first byte of a valid path is
```

Comment:
you can simply check ``os.path.exists(path_or_buf)`` (see how this is done in ``pandas.io.json.json.read_json``